src/Text/Pandoc/PDF.hs

```diff
@@ -276,12 +274,11 @@ runTeXProgram verbosity program args runNumber numRuns tmpDir' source = do
     let file = tmpDir </> "input.tex"
     exists <- liftIO $ doesFileExist file
     unless exists $ liftIO $ BS.writeFile file $ UTF8.fromText source
-    let programArgs =
-          if takeBaseName program == "latexmk"
-             then ["-interaction=batchmode", "-halt-on-error", "-pdf",
-                   "-quiet", "-outdir=" ++ tmpDir] ++ args ++ [file]
-             else ["-halt-on-error", "-interaction", "nonstopmode",
-                   "-output-directory", tmpDir] ++ args ++ [file]
+    let programArgs | takeBaseName program == "latexmk"  = ["-interaction=batchmode", "-halt-on-error", "-pdf",
+                                                            "-quiet", "-outdir=" ++ tmpDir] ++ args ++ [file]
+                    | takeBaseName program == "tectonic" = args ++ [file] -- outdir defaults to dir of input file
```

Comment:
One question is what tectonic will put in the output dir.  The pandoc policy is not to leave any files on the file system that the user didn't explicitly request (e.g. with `-o myfile.pdf`).

Reply:
The answer to that one is nothing.

It has extra flags that optionally enable writing aux files into the output dir (which is, as the comment says, the temp directory in this case), but it never needs them, that’s only so you can speed up reprocessing with the Makefiles it also knows how to generate.